### PR TITLE
all tests for PATCH articles by arcticle id pass

### PR DESCRIPTION
**Pull request by baddis01**
all tests pass for ticket #7

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -84,6 +84,45 @@ describe("app", () => {
...
+        .send({ inc_votes: "twelve" })
+        .expect(400)
+        .then((res) => {
+          expect(res.body.msg).toBe("Bad Request");
+        });
+    });
+    test("status: 400 - should return with 'Bad Request' when key is wrong", () => {
+      return request(app)
+        .patch("/api/articles/4")
+        .send({ increase_das_votes: 6 })
```
Comment (dav3rid) [inline __tests__/app.test.js:128]: haha

File: __tests__/app.test.js [hunk 2]
```diff
@@ -84,6 +84,45 @@ describe("app", () => {
...
+  describe("PATCH - /api/articles/:article_id", () => {
+    test("status: 202 - should accept the request in the form of an object { inc_vote: newVote }", () => {
+      return request(app)
+        .patch("/api/articles/1")
+        .send({ inc_votes: 10 })
+        .expect(201)
+        .then((res) => {
+          expect(res.body.article.votes).toBe(110);
+        });
+    });
```
Comment (dav3rid) [inline __tests__/app.test.js:97]: You can have another happy path test to ensure negative values are still accepted

Comment (dav3rid) [inline __tests__/app.test.js:97]: You can have another happy path test to ensure negative values are still accepted

File: __tests__/app.test.js [hunk 3]
```diff
@@ -84,6 +84,45 @@ describe("app", () => {
...
+    test("status: 400 - should return with 'Bad Request' when key is wrong", () => {
+      return request(app)
+        .patch("/api/articles/4")
+        .send({ increase_das_votes: 6 })
+        .expect(400)
+        .then((res) => {
+          expect(res.body.msg).toBe("Bad Request - Missing/Incorrect Fields");
+        });
+    });
+  });
```
Comment (dav3rid) [inline __tests__/app.test.js:152]: A couple more sad path tests would be of use here - very similar to ones on your GET endpoint
- 404 article id not found
- 400 article id invalid
Both with a correctly formed request body

File: __tests__/app.test.js [hunk 4]
```diff
@@ -84,6 +84,45 @@ describe("app", () => {
         });
     });
   });
+
+  describe("PATCH - /api/articles/:article_id", () => {
+    test("status: 202 - should accept the request in the form of an object { inc_vote: newVote }", () => {
+      return request(app)
+        .patch("/api/articles/1")
+        .send({ inc_votes: 10 })
+        .expect(201)
```
Comment (dav3rid) [inline __tests__/app.test.js]: Your test description says 202 and you are asserting for a 201...
I would actually say a good old 200 is appropriate here - but a 202 "accepted" not inappropriate either
We would generally reserve 201 for the creation of a brand new resource rather than amendment

File: controllers/errors.controller.js [hunk 5]
```diff
@@ -7,5 +7,7 @@ exports.handleCustomErrors = (err, req, res, next) => {
 exports.handlePsqlErrors = (err, req, res, next) => {
   if (err.code === "22P02") {
     res.status(400).send({ msg: "Bad Request" });
+  } else if (err.code === "23502") {
+    res.status(400).send({ msg: err.msg || "Bad Request" });
```
Comment (dav3rid) [inline controllers/errors.controller.js]: You'll find that the error will only have a `msg` property if it's one of your custom Promise rejections. I'd have thought you'll be able to hardcode "Bad Request" here as the above error handler will have already handled this 👍

**Review by dav3rid — COMMENTED** [review]
Well done!

File: __tests__/app.test.js [hunk 6]
```diff
@@ -86,7 +86,7 @@ describe("app", () => {
   });
 
   describe("PATCH - /api/articles/:article_id", () => {
-    test("status: 202 - should accept the request in the form of an object { inc_vote: newVote }", () => {
+    test("status: 201 - should accept the request in the form of an object { inc_vote: newVote }", () => {
```
Comment (dav3rid) [inline __tests__/app.test.js]: I would have actually gone the other way here Ben - asserting for a 200 or 202 will be more appropriate than a 201.
_**official**_ docs: https://http.cat/

File: __tests__/app.test.js [hunk 7]
```diff
@@ -86,15 +86,24 @@ describe("app", () => {
   });
 
   describe("PATCH - /api/articles/:article_id", () => {
-    test("status: 201 - should accept the request in the form of an object { inc_vote: newVote }", () => {
+    test("status: 200 - should accept the request in the form of an object { inc_vote: newVote } taking a positive number", () => {
```
Comment (dav3rid) [inline __tests__/app.test.js:89]: Oh sorry - you have!!! Well done!

File: __tests__/app.test.js [hunk 8]
```diff
@@ -86,15 +86,24 @@ describe("app", () => {
...
     });
+    test("status: 200 - should accept the request in the form of an object { inc_vote: newVote } taking a negative number", () => {
+      return request(app)
+        .patch("/api/articles/1")
+        .send({ inc_votes: -10 })
+        .expect(200)
+        .then((res) => {
+          expect(res.body.article.votes).toBe(90);
+        });
+    });
```
Comment (dav3rid) [inline __tests__/app.test.js:106]: 🎉

**Review by dav3rid — APPROVED** [review]
